app/cli/startstream.cpp

```diff
@@ -0,0 +1,190 @@
+#include "startstream.h"
+#include "backend/computermanager.h"
+#include "streaming/session.hpp"
+
+#include <QTimer>
+
+#define COMPUTER_SEEK_TIMEOUT 15000
+#define APP_SEEK_TIMEOUT 15000
+
+namespace CliStartStream
+{
+
+enum State {
+    StateInit,
+    StateSeekComputer,
+    StateSeekApp,
+    StateStartSession,
+    StateFailure,
+};
+
+class Event
+{
+public:
+    enum Type {
+        ComputerUpdated,
+        Executed,
+        Timedout,
+    };
+
+    Event(Type type)
+        : type(type), computerManager(nullptr), computer(nullptr) {}
+
+    Type type;
+    ComputerManager *computerManager;
+    NvComputer *computer;
+};
+
+class LauncherPrivate
+{
+    Q_DECLARE_PUBLIC(Launcher)
+
+public:
+    LauncherPrivate(Launcher *q) : q_ptr(q) {}
+
+    void handleEvent(Event event)
+    {
+        Q_Q(Launcher);
+        Session* session;
+        NvApp app;
+
+        switch (event.type) {
+        case Event::Executed:
+            if (m_State == StateInit) {
+                m_State = StateSeekComputer;
+                m_TimeoutTimer->start(COMPUTER_SEEK_TIMEOUT);
+                m_ComputerManager = event.computerManager;
+                q->connect(m_ComputerManager, &ComputerManager::computerStateChanged,
+                           q, &Launcher::onComputerUpdated);
```

Comment:
Connecting `ComputerManager::computerAddCompleted()` would allow you to catch an unreachable host before the 15 second timeout

Reply:
Umm, there is also the mDNS polling going on. I assume that the `computerAddCompleted()` would not tell if mDNS can't find it, right?

Reply:
It would only be emitted if the manual addComputer() fails. However, it now occurs to me that doing that would break resolving UUIDs (since add will fall for those).

I still don't like having to wait 15 seconds when the user mistypes an IP address or hostname. Maybe you could lower the timeouts from 15 to 10 seconds as a compromise?

Reply:
Yeah, that is fine for me. When I tested this, both the manual `addComputer()` and the mDNS quieries found the host computer almost instantiously. But, I was unsure if my experience was normal or not. So, I just picked some high-ish enough numbers as timeout values.

Reply:
I'll lower the app list loading timeout to 10 too.